### PR TITLE
New metrics dashboard for v2

### PR DESCRIPTION
**Pull request by kerneltime**
## Description
New dashboard for v2 metrics.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by kerneltime** [comment]
image

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#d3893882-76cc-469c-a5c7-a288f150141e) |
<a name="d3893882-76cc-469c-a5c7-a288f150141e"></a>
### 11739-1dafe8f/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30964
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7f371808cfb2:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2594,
  "function": "aws --endpoint-url http://minio-c3.minio.io:30964 s3api copy-object --bucket awscli-mint-test-bucket-674 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-674/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 49 seconds
(7/15) Running minio-dotnet tests ... done in 40 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 23 seconds
{
  "args": {},
  "duration": 465,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 29 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 422,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-57a481, objectName = minio-java-test-mdnarl-copy, resource = /minio-java-test-57a481/minio-java-test-mdnarl-copy, requestId = 166A9BD9C1B714B4, hostId = 50905d26-a6ea-46dc-b82b-07b52539da68)\nrequest={method=PUT, url=http://minio-c3.minio.io:30964/minio-java-test-57a481/minio-java-test-mdnarl-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-38rrma7/minio-java-test-mdnarl\nHost: minio-c3.minio.io:30964\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210309T072615Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210309/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 414\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Tue, 09 Mar 2021 07:26:15 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 166A9BD9C1B714B4\nX-Xss-Protection: 1; mode=block\nDate: Tue, 09 Mar 2021 07:26:15 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 1 minutes and 5 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 3 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2907",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-32531/s3cmd-test-object-17687 6bcb71e8-c67b-4328-8fec-1dcbd06d38b2\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 6bcb71e8-c67b-4328-8fec-1dcbd06d38b2 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 8 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#588ef436-ad52-4858-8f65-ea83c236565d) |
<a name="588ef436-ad52-4858-8f65-ea83c236565d"></a>
### 11739-ba21c4f/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31582
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d668828716b5:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 3 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... FAILED in 33 seconds
{
  "name": "awscli",
  "duration": 2658,
  "function": "aws --endpoint-url http://minio-dev8.minio.io:31582 s3api copy-object --bucket awscli-mint-test-bucket-523 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-523/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 47 seconds
(7/15) Running minio-dotnet tests ... done in 45 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 57 seconds
{
  "args": {},
  "duration": 507,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 53 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 336,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-2onu8jc, objectName = minio-java-test-2dl1phq-copy, resource = /minio-java-test-2onu8jc/minio-java-test-2dl1phq-copy, requestId = 166A9D0C516F4157, hostId = b877f8e6-349d-4fc5-b247-84d7c69326c2)\nrequest={method=PUT, url=http://minio-dev8.minio.io:31582/minio-java-test-2onu8jc/minio-java-test-2dl1phq-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-1blemqo/minio-java-test-2dl1phq\nHost: minio-dev8.minio.io:31582\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210309T074810Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210309/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Tue, 09 Mar 2021 07:48:12 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 166A9D0C516F4157\nX-Xss-Protection: 1; mode=block\nDate: Tue, 09 Mar 2021 07:48:12 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 50 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 40 seconds
(10/15) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "3310",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-3157/s3cmd-test-object-31806 05ee18af-0f57-413f-b5de-8f8a1191619f\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 05ee18af-0f57-413f-b5de-8f8a1191619f put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 7 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by kerneltime** [comment]
Fixes #11543

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Can you get a writeoff on this from @abperiasamy regarding the UI this is what we want to show?

**Comment by harshavardhana** [comment]
So I installed this dashboard here @kerneltime https://play.min.io:3000/d/eTOSPxYGz/metrics-v2?orgId=1 and it looks like disks offline and node offline are giving super incorrect results. 

Looks like your query is wrong.

**Comment by apiszcz** [comment]
I have tried 4 other json grafana dashboards all appear to be based on an older set of metrics.

**Comment by harshavardhana** [comment]
> I have tried 4 other json grafana dashboards all appear to be based on an older set of metrics.

This is not done yet @apiszcz please use the older dashboard for now.

**Comment by niphlod** [comment]
jm2c: keeping as close as possible to the "old" dashboard https://gist.github.com/niphlod/92608c184b92de773cbf70e85f38970b .

Remaining doubts: minio_s3_requests_inflight_total vs minio_s3_requests_waiting_total for "Active S3 requests" . Everything else maps more or less the same (I just changed metric name and labels accordingly)

Something is not adding up with the "minio_cluster_disk_offline_total", my setup (single node, no cluster, RELEASE.2021-03-17T02-33-02Z, win64) shows 1 even if there are no offline disks.
Summing up, grafana calcs seems to be fine, but ATM
minio_cluster_disk_total = 1
minio_cluster_disk_online_total = 0
minio_cluster_disk_offline_total = 1
in a simple setup with one disk (e.g. minio.exe server --address :3000 C:\path).
I feel that *maybe* online and offline are "switched" by minio for a "bug".

**Comment by krejcmat** [comment]
I am having the same problem as @niphlod. Single server setup with one disk and getting as well:
minio_cluster_disk_total = 1
minio_cluster_disk_online_total = 0
minio_cluster_disk_offline_total = 1

Linux Ubuntu 18.04 RELEASE.2021-04-06T23-11-00Z

**Comment by harshavardhana** [comment]
This PR will be resubmitted with more changes. @krejcmat please open another Github issue that doesn't look right but its unrelated to this PR.
